qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModelUtils.h

```diff
@@ -0,0 +1,124 @@
+
+bool doesExistInADS(std::string const &workspaceName) {
+  return AnalysisDataService::Instance().doesExist(workspaceName);
+}
+
+MatrixWorkspace_sptr getADSMatrixWorkspace(std::string const &workspaceName) {
+  return AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(workspaceName);
+}
+
+WorkspaceGroup_sptr getADSWorkspaceGroup(std::string const &workspaceName) {
+  return AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(workspaceName);
+}
+
+IAlgorithm_sptr loadAlgorithm(std::string const &filename, std::string const &outputName) {
+  auto loader = AlgorithmManager::Instance().create("Load");
+  loader->initialize();
+  loader->setProperty("Filename", filename);
+  loader->setProperty("OutputWorkspace", outputName);
+  return loader;
+}
+
+std::string createRangeString(std::size_t const &from, std::size_t const &to) {
+  return std::to_string(from) + "-" + std::to_string(to);
+}
+
+std::string createGroupString(std::size_t const &start, std::size_t const &size) {
+  return createRangeString(start, start + size - 1);
+}
+
+std::string createGroupingString(std::size_t const &groupSize, std::size_t const &numberOfGroups,
+                                 std::size_t const &spectraMin) {
+  auto groupingString = createRangeString(spectraMin, spectraMin + groupSize - 1);
+  for (auto i = spectraMin + groupSize; i < spectraMin + groupSize * numberOfGroups; i += groupSize)
+    groupingString += "," + createGroupString(i, groupSize);
+  return groupingString;
+}
+
+std::string createDetectorGroupingString(std::size_t const &groupSize, std::size_t const &numberOfGroups,
+                                         std::size_t const &numberOfDetectors, std::size_t const &spectraMin) {
+  const auto groupingString = createGroupingString(groupSize, numberOfGroups, spectraMin);
+  const auto remainder = numberOfDetectors % numberOfGroups;
+  if (remainder == 0)
+    return groupingString;
+  return groupingString + "," +
+         createRangeString(spectraMin + numberOfDetectors - remainder, spectraMin + numberOfDetectors - 1);
+}
+
+std::string createDetectorGroupingString(std::size_t const &numberOfDetectors, std::size_t const &numberOfGroups,
+                                         std::size_t const &spectraMin) {
+  const auto groupSize = numberOfDetectors / numberOfGroups;
+  if (groupSize == 0)
+    return createRangeString(spectraMin, spectraMin + numberOfDetectors - 1);
+  return createDetectorGroupingString(groupSize, numberOfGroups, numberOfDetectors, spectraMin);
+}
+
+void deleteWorkspace(std::string const &name) {
+  auto deleter = AlgorithmManager::Instance().create("DeleteWorkspace");
+  deleter->initialize();
+  deleter->setProperty("Workspace", name);
+  deleter->execute();
+}
+
+double getSampleLog(const MatrixWorkspace_const_sptr &workspace, std::string const &logName,
+                    double const &defaultValue) {
+  try {
+    return workspace->getLogAsSingleValue(logName);
```

Comment:
As discussed the two functions of this name (`getSampleLog`) could be refactored to be clearer (we think it is looking for log value that could have one of 3 names. I think it should not catch an exception but check it exists using `hasProperty` and then could even throw a warning if none of the logs exist and a default value is being used.
@AnthonyLim23 suggested this would make a good induction issue, there may be other similar functions in here too.

Reply:
I've refactored the function in the current PR because it is a trivial refactoring